### PR TITLE
Add ToTwitterFuture type class

### PR DESCRIPTION
**Pull request by ImLiar**
Type class to support different async mechanisms. There is an identity for twitter future and scala futures support in different package.

Separate PR from https://github.com/finagle/finch/pull/816

### CODE REVIEW

**Comment by ImLiar** [comment]
Summon @vkostyukov @rpless

File: core/src/main/scala/io/finch/syntax/ToTwitterFuture.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package io.finch.syntax
+
+import cats.Monad
+import com.twitter.util.Future
+import io.catbird.util._
+
+/**
+  * Type class for conversion of some HKT (i.e. scala.concurrent.Future) to twitter future
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/syntax/ToTwitterFuture.scala]: Wrap `s.c.Future` with ticks please.

Comment (vkostyukov) [reply core/src/main/scala/io/finch/syntax/ToTwitterFuture.scala]: Also s/twitter future/Twitter Future.

File: core/src/main/scala/io/finch/syntax/ToTwitterFuture.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  */
+trait ToTwitterFuture[F[_]] {
+
+  def M: Monad[F]
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/syntax/ToTwitterFuture.scala]: Why is this necessary?

File: core/src/test/scala/io/finch/syntax/MapperSyntaxBehaviour.scala [hunk 3]
```diff
@@ -0,0 +1,104 @@
+package io.finch.syntax
+
+import cats.syntax.functor._
+import com.twitter.finagle.http.Response
+import io.finch._
+import io.finch.{FinchSpec, Text}
+import org.scalacheck.Arbitrary
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+
+trait MapperSyntaxBehaviour extends FinchSpec with GeneratorDrivenPropertyChecks {
```
Comment (vkostyukov) [inline core/src/test/scala/io/finch/syntax/MapperSyntaxBehaviour.scala]: I think we name these `XSpec`?

File: core/src/main/scala/io/finch/syntax/ToTwitterFuture.scala (hunk 2 above)
Comment (ImLiar) [reply core/src/main/scala/io/finch/syntax/ToTwitterFuture.scala]: To make generic testing via behaviour possible

File: core/src/test/scala/io/finch/syntax/MapperSyntaxBehaviour.scala (hunk 3 above)
Comment (ImLiar) [reply core/src/test/scala/io/finch/syntax/MapperSyntaxBehaviour.scala]: It's not a `Spec` strictly speaking, it's a trait with behaviour method

**Comment by rpless** [comment]
LGTM. Great work @ImLiar!

File: core/src/main/scala/io/finch/syntax/ToTwitterFuture.scala [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import cats.Monad
+import com.twitter.util.Future
+import io.catbird.util._
+
+/**
+  * Type class for conversion of some HKT (i.e. `scala.concurrent.Future`) to `com.twitter.util.Future`
+  */
+trait ToTwitterFuture[F[_]] {
+
+  def M: Monad[F]
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/syntax/ToTwitterFuture.scala]: Can we derive it in tests (if it's the only place it's used) instead?

**Comment by ImLiar** [comment]
@vkostyukov done

File: core/src/test/scala/io/finch/syntax/MapperSyntaxBehaviour.scala (hunk 3 above)
Comment (vkostyukov) [reply core/src/test/scala/io/finch/syntax/MapperSyntaxBehaviour.scala]: Sure. But we have zero tests named `Behaviour` and something like 30 tests named `Spec`. Maybe we can rename it for the sake of consistency?

**Comment by ImLiar** [comment]
@vkostyukov done

**Comment by vkostyukov** [comment]
Thanks again, @ImLiar! Really nice work!
